src/web/app/pages-student/student-course-details-page/student-course-details-page.component.ts

```diff
@@ -114,17 +114,18 @@ export class StudentCourseDetailsPageComponent implements OnInit {
             this.studentProfileService.getStudentProfile(student.email, courseId)
                   .subscribe((studentProfile: StudentProfile) => {
                     const newPhotoUrl: string =
-              `${environment.backendUrl}/webapi/student/profilePic?courseid=${courseId}&studentemail=${student.email}`;
+                      `${environment.backendUrl}/webapi/student/profilePic`
+                      + `?courseid=${courseId}&studentemail=${student.email}`;
 
                     const newTeammateProfile: StudentProfileWithPicture = {
-                      studentProfile,
+                      // Use student.email as studentProfile.email is the long term email which should not be shown
```

Comment:
Correct me if I am wrong. Looks like from your conversation history this is a potential security glitch says that "the long term email should not be shown to student". If this is the case, doing override in the frontend won't work. We need hide the field from the backend.

Reply:
Yea you're right. Since this got larger in scale I'll KIV this first and return after the search migration API epic